### PR TITLE
[windows] conditionally add windows cws components.

### PR DESCRIPTION
**Pull request by derekwbrown**
### What does this PR do?

Adds Windows CWS agent to windows installer conditionally.

Components are added only based on inclusion in `release.json`.
Components are included in nightly builds, but not in release builds

### Motivation

Include under development components in nightly builds to facilitate ongoing development without including in release builds

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [x] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [x] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Comment by clarkb7** [comment]
does `security-agent.exe` need to be added to `additional_sign_files_list` so it gets signed?

File: omnibus/config/projects/agent.rb
Comment (chouquette) [inline omnibus/config/projects/agent.rb:234]: :question: not directly related to this PR, but do we still support Windows 32bits?

Comment (chouquette) [inline omnibus/config/projects/agent.rb:234]: Couldn't you check if the file exists rather than probing the environment? If I understand your change correctly, `security-agent.exe` will only be there if it was built, based on the same condition in `datadog-agent.rb`
This would be simpler to read and to toggle IMO

File: omnibus/config/projects/agent.rb
Comment (chouquette) [inline omnibus/config/projects/agent.rb:421]: Same question about checking the file existence rather than the flag

File: omnibus/config/software/datadog-agent-finalize.rb
Comment (chouquette) [inline omnibus/config/software/datadog-agent-finalize.rb:31]: Same question

**Review by chouquette — APPROVED** [review]
LGTM for agent-platform files, with non-blocking comments

**Review by paulcacheux — APPROVED** [review]

**Review by safchain — APPROVED** [review]

File: pkg/config/system_probe_cws_windows.go
Comment (paulcacheux) [inline pkg/config/system_probe_cws_windows.go:1]: Can you add this file and the `notwin` version to the CODEOWNERS file (owned by our team) ? So that it matches the rest of our config

Comment (derekwbrown) [reply pkg/config/system_probe_cws_windows.go:1]: yup.  And renamed the confusing named `windows_notwin.go` file too

**Review by ISauve — APPROVED** [review]

**Review by paulcacheux — APPROVED** [review]

File: pkg/config/config_template.yaml
Comment (buraizu) [inline pkg/config/config_template.yaml:1934]:
```suggestion
    ## Path from where the policy files are loaded
```

**Review by buraizu — APPROVED** [review]
Thanks, approving with a minor suggestion

**Review by clarkb7 — APPROVED** [review]

File: pkg/config/config_template.yaml
Comment (dustmop) [inline pkg/config/config_template.yaml:1920]: The `@param` above is `sysprobe_socket`, which doesn't match here or the usages in `pkg/config/system_probe_cws_windows.go`. Which is correct?

**Review by dustmop — APPROVED** [review]
LGTM for agent-shared-components, with one question

File: pkg/config/config_template.yaml
Comment (derekwbrown) [reply pkg/config/config_template.yaml:1920]: sysprobe socket & socket are different; socket is for the cws component.
However, I have a note (for a separate pr) that the system probe config needs some windows-specific updating.

**Pull request merged**
